telemetry_core/src/main/java/com/newrelic/telemetry/AbstractSenderBuilder.java

```diff
@@ -0,0 +1,112 @@
+package com.newrelic.telemetry;
+
+import com.newrelic.telemetry.http.HttpPoster;
+import com.newrelic.telemetry.util.Utils;
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+
+public abstract class AbstractSenderBuilder<B extends AbstractSenderBuilder<B>> {
```

Comment:
I get a twitchy when I see abstract parameterized builders. Especially when the `build()` method ends up returning `Object`. Let's create an issue, to be done quickly after this is merged, get rid of the templated builder and replace it with something better. I would bet that @breedx-nr has all the solutions to making it great in his back pocket.

Reply:
#139 filed to track this.